code/game/machinery/washing_machine.dm

```diff
@@ -222,14 +307,28 @@
 		else
 			return ..()
 	else if(istype(W,/obj/item/grab))
-		if( (state == 1) && hacked)
+		if(open && !full)
 			var/obj/item/grab/G = W
-			if(ishuman(G.assailant) && iscorgi(G.affecting))
-				G.affecting.loc = src
-				qdel(G)
-				state = 3
+			if((ishuman(G.assailant) && isanimal(G.affecting) && !ishostile(G.affecting) && !isbot(G.affecting)) || (ishuman(G.assailant) && ishuman(G.affecting) && HAS_TRAIT(G.affecting, TRAIT_DWARF)))
+				if(ishuman(G.assailant) && ishuman(G.affecting) && HAS_TRAIT(G.affecting, TRAIT_DWARF) && (G.state < GRAB_AGGRESSIVE))
+					to_chat(user, "<span class='warning'>You need a better grip to do that!</span>")
+				else
+					visible_message("[user] starts putting [G.affecting.name] into [src].")
+					if(do_after(user, 20, target = G.affecting))
+						visible_message("[user] puts [G.affecting.name] into [src].")
+						add_fingerprint(user)
+						G.affecting.loc = src
+						qdel(G)
+						full = TRUE
 			update_icon()
 		else
+			if(!open)
+				to_chat(user, "<span class='notice'>[src]'s door is closed.</span>")
+			else
+				if(full)
+					to_chat(user, "<span class='notice'>[src] is already full.</span>")
+				else
+					to_chat(user, "<span class='notice'>You can't fit [W] in [src].</span>")
 			return ..()
 	else if(istype(W,/obj/item/stack/sheet/hairlesshide) || \
```

Comment:
this is actually a horrific else if check, use a list of typepaths and check if W's type is in it.